api/app/Console/Commands/RoleSync.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Console\Command;
+use App\Models\Role;
+use App\Models\Team;
+use App\Models\User;
+use Database\Helpers\ApiEnums;
+
+class RoleSync extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'role:sync';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Sync Laratrust roles to existing users.';
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $baseUserRole = Role::where('name', 'base_user')->sole();
+        $applicantRole = Role::where('name', 'applicant')->sole();
+        $poolOperatorRole = Role::where('name', 'pool_operator')->sole();
+        $requestResponderRole = Role::where('name', 'request_responder')->sole();
+        $superAdminRole = Role::where('name', 'platform_admin')->sole();
+
+        $DCMTeam = Team::where('name', 'digital-community-management')
+            ->sole();
+
+        $users = User::all();
+
+        foreach($users as $user) {
+            // Only sync roles if they have none assigned
```

Comment:
I think we want to be able to run this periodically to ensure that the new roles stay in sync with the old roles as the old roles slowly change with use.  If my user gets migrated and then my legacy_role changes this script will not sync me back up, right?
> This command should be safe to run multiple times in production, as user LegacyRoles may change.

Reply:
Okay, so we are okay with overwriting existing roles? I did this guard in case of the following scenario:

- User has legacy role `APPLICANT`, user was assigned the new role `request_responder`. 
- Command is run
- sees `APPLICANT` legacy role
- syncs new `base_user` and `applicant` role
- sync removes the `request_responder` role

My interpretation of "safe to run multiple times" was that we could run it without overwriting users with the new roles and only affect those with legacy roles, updating them to the new roles once.

Reply:
My understanding is that this sync command is the only way a user will get new roles until we "cut over" so we won't have to worry about overwriting a new role that was manually added.  Worth bringing up in scrum today to confirm, though.

Reply:
We discussed this in scrum today.  My understanding is correct - this command should overwrite whatever "new roles" the user has to sync them up from "old roles" freshly.  Once we cutover to managing new roles directly we will stop using this command.

Reply:
Thanks, made that change in [update to overwrite roles on each run](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5795/commits/2ae3e4bb619289131d53f09e3264fce889bdfbf5)